examples/failure-policy/host-maintenance-event-model.yaml

```diff
@@ -0,0 +1,71 @@
+apiVersion: jobset.x-k8s.io/v1alpha2
+kind: JobSet
+metadata:
+  name: host-maintenance-event-model
+spec:
+  failurePolicy:
+    maxRestarts: 0
+    rules:
+      # The JobSet will restart an unlimited number of times when failure matches the pod failure policy.
+      - action: RestartJobSetAndIgnoreMaxRestarts
+        onJobFailureReasons:
+        - PodFailurePolicy
+      # The JobSet is restarted as normal when the leader job fails and the above rule is not matched.
```

Comment:
What does it mean for this to restart with `maxRestarts 0`? It would fail right away right?